source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -416,19 +416,19 @@ void OAuth2Filter::onGetAccessTokenSuccess(const std::string& access_code,
   id_token_ = id_token;
   refresh_token_ = refresh_token;
 
-  const auto new_epoch = time_source_.systemTime() + expires_in;
-  new_expires_ = std::to_string(
-      std::chrono::duration_cast<std::chrono::seconds>(new_epoch.time_since_epoch()).count());
+  new_expires_ = time_source_.systemTime() + expires_in;
```

Comment:
It would be better just to pass the `expires_in` to the `finishFlow()` to avoid re-calculate the MaxAge

Reply:
Done.